Sources/MapboxNavigation/NavigationMapView+BuildingHighlighting.swift

```diff
@@ -11,17 +11,96 @@ extension NavigationMapView {
      
      - parameter coordinates: Coordinates which represent building locations.
      - parameter extrudesBuildings: Switch which allows to highlight buildings in either 2D or 3D. Defaults to true.
-     
-     - returns: Bool indicating if number of buildings found equals number of coordinates supplied.
+     - parameter completion: An escaping closure that operates after the buildings are highted.
+
      */
-    @discardableResult public func highlightBuildings(at coordinates: [CLLocationCoordinate2D], in3D extrudesBuildings: Bool = true) -> Bool {
-        return false
+    public func highlightBuildings(at coordinates: [CLLocationCoordinate2D],
+                                   in3D extrudesBuildings: Bool = true,
+                                   completion: @escaping (_ result: Bool) -> Void) {
+        var foundBuildingIds = Set<Double>()
+        let group = DispatchGroup()
+        if let identifiers = try? mapView.__map.getStyleLayers().compactMap({$0.id}).filter({ $0.contains("building") }) {
+            let _ = coordinates.map({
+                group.enter()
+                let screenCoordinate = mapView.point(for: $0, in: self)
+                mapView.visibleFeatures(at: screenCoordinate,
+                                        styleLayers: Set(identifiers),
+                                        completion: { [weak self] result in
+                                            guard let validSelf = self else { return }
+                                            if case .success(let features) = result {
+                                                if let identifier = features.first?.featureIdentifierToDouble() {
+                                                    foundBuildingIds.insert(identifier)
+                                                }
+                                                group.leave()
+                                            }})
+            })
+
+            group.notify(queue: DispatchQueue.global()) {
+                self.addBuildingsLayer(with: foundBuildingIds, in3D: extrudesBuildings)
+                completion(foundBuildingIds.count == coordinates.count)
+            }
+        } else {
+            completion(false)
+        }
     }
     
     /**
      Removes the highlight from all buildings highlighted by `highlightBuildings(at:in3D:)`.
      */
     public func unhighlightBuildings() {
-        
+        guard let _ = try? mapView.style.getLayer(with: IdentifierString.buildingExtrusionLayer, type: FillExtrusionLayer.self).get() else { return }
+        mapView.style.removeStyleLayer(forLayerId: IdentifierString.buildingExtrusionLayer)
```

Comment:
Calls to this method produces warning if result is not handled. Please handle result even if it's not used so that we do not get any warnings.

Reply:
Thanks for the suggestion! I'll add the result part.